src/ImageSharp/Formats/ImageExtesions.Save.cs

```diff
@@ -0,0 +1,694 @@
+        public static void SaveAsBmp(this Image source, string path, BmpEncoder encoder) =>
+            source.Save(
+                path,
+                encoder ?? source.GetConfiguration().ImageFormatsManager.FindEncoder(BmpFormat.Instance));
+
+        /// <summary>
+        /// Saves the image to the given stream with the Bmp format.
+        /// </summary>
+        /// <param name="source">The image this method extends.</param>
+        /// <param name="path">The file path to save the image to.</param>
+        /// <param name="encoder">The encoder to save the image with.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown if the path is null.</exception>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public static Task SaveAsBmpAsync(this Image source, string path, BmpEncoder encoder)
+            => SaveAsBmpAsync(source, path, encoder, default);
+
+        /// <summary>
+        /// Saves the image to the given stream with the Bmp format.
+        /// </summary>
+        /// <param name="source">The image this method extends.</param>
+        /// <param name="path">The file path to save the image to.</param>
+        /// <param name="encoder">The encoder to save the image with.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown if the path is null.</exception>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public static Task SaveAsBmpAsync(this Image source, string path, BmpEncoder encoder, CancellationToken cancellationToken) =>
+            source.SaveAsync(
+                path,
+                encoder ?? source.GetConfiguration().ImageFormatsManager.FindEncoder(BmpFormat.Instance),
+                cancellationToken);
+
+        /// <summary>
+        /// Saves the image to the given stream with the Bmp format.
+        /// </summary>
+        /// <param name="source">The image this method extends.</param>
+        /// <param name="stream">The stream to save the image to.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown if the stream is null.</exception>
+        public static void SaveAsBmp(this Image source, Stream stream)
+            => SaveAsBmp(source, stream, null);
+
+        /// <summary>
+        /// Saves the image to the given stream with the Bmp format.
+        /// </summary>
+        /// <param name="source">The image this method extends.</param>
+        /// <param name="stream">The stream to save the image to.</param>
+        /// <exception cref="System.ArgumentNullException">Thrown if the stream is null.</exception>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public static Task SaveAsBmpAsync(this Image source, Stream stream) => SaveAsBmpAsync(source, stream, null);
```

Comment:
I'm torn here.... We've been very explicit in avoiding default parameters throughout the codebase. 

My OCD vs the might of the BCL ... ooft!

Reply:
I generally agree on default parameters but in my head cancellation tokens are a different beast to most default params... also and without it we end up doubling our async API surface.

Reply:
See my previous comment. I also like the `CancellationToken ct = default` API shape much better in general, but I'm afraid it's too late.

Reply:
Note: BCL also has many places with multiple explicit overloads instead of `= default`. The reason for that was the same: to avoid breaking the previous non-cancellable overloads.

Reply:
its not too late we haven't release a v1 yet